### PR TITLE
Adding postgres  configuration component in E2E test

### PR DESCRIPTION
**Pull request by shivamkm07**
# Description

This PR adds postgres configuration component to e2e test

## Issue reference

Please reference the issue this PR will close: #6230 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by shivamkm07** [comment]
Note that test here for postgres component fails because it requires a fix in the component included in PR https://github.com/dapr/components-contrib/pull/2740. So it will only pass after the above PR is merged and go.mod points to the updated master in contrib.

**Comment by yaron2** [comment]
Why is this needed? We do not test individual components in `dapr/dapr`.

**Comment by shivamkm07** [comment]
@yaron2 As per https://github.com/dapr/proposals/blob/main/guides/api-design.md#beta-to-stable, `
E2E tests must exercise at least two different implementations of the building block's API`. Currently only `redis` is being tested in E2E test. This PR is to add postgres as well to the same.

**Comment by yaron2** [comment]
> @yaron2 As per https://github.com/dapr/proposals/blob/main/guides/api-design.md#beta-to-stable, ` E2E tests must exercise at least two different implementations of the building block's API`. Currently only `redis` is being tested in E2E test. This PR is to add postgres as well to the same. cc @mukundansundar

I'll work to remove this requirement, thanks for bringing it up. It makes no sense to test multiple components from the API perspective: if one component test fails, it means there's an implementation issue with that component and we have conformance/certification tests to ensure the validity from the components side.

Can you think of any practical example where testing a minimum of 2 components (out of possibly dozens) would yield real benefits from testing them in dapr/dapr e2e?

**Comment by mukundansundar** [comment]
> > @yaron2 As per https://github.com/dapr/proposals/blob/main/guides/api-design.md#beta-to-stable, ` E2E tests must exercise at least two different implementations of the building block's API`. Currently only `redis` is being tested in E2E test. This PR is to add postgres as well to the same. cc @mukundansundar
> 
> I'll work to remove this requirement, thanks for bringing it up. It makes no sense to test multiple components from the API perspective: if one component test fails, it means there's an implementation issue with that component and we have conformance/certification tests to ensure the validity from the components side.
> 
> Can you think of any practical example where testing a minimum of 2 components (out of possibly dozens) would yield real benefits from testing them in dapr/dapr e2e?

@yaron2 We have had scenarios where sometimes having a second E2E component has helped. For example with respect to CosmosDB and state store building block. 

For the question of whether it would help in this particular scenario, it might.
One particular thing here is should we have only Redis for most of the building blocks or should we have other components also?

**Comment by yaron2** [comment]
> > > @yaron2 As per https://github.com/dapr/proposals/blob/main/guides/api-design.md#beta-to-stable, ` E2E tests must exercise at least two different implementations of the building block's API`. Currently only `redis` is being tested in E2E test. This PR is to add postgres as well to the same. cc @mukundansundar
> > 
> > 
> > I'll work to remove this requirement, thanks for bringing it up. It makes no sense to test multiple components from the API perspective: if one component test fails, it means there's an implementation issue with that component and we have conformance/certification tests to ensure the validity from the components side.
> > Can you think of any practical example where testing a minimum of 2 components (out of possibly dozens) would yield real benefits from testing them in dapr/dapr e2e?
> 
> @yaron2 We have had scenarios where sometimes having a second E2E component has helped. For example with respect to CosmosDB and state store building block.
> 
> For the question of whether it would help in this particular scenario, it might. One particular thing here is should we have only Redis for most of the building blocks or should we have other components also?

I'm interested in knowing how this can help detect runtime API issues instead of component implementation issues. If the purpose is to help find issues in components implementations, then `dapr/dapr` is not the place as this responsibility falls directly under contrib conformance and certification tests. More so, this feels heavily non-deterministic - if you have N components but only pick 2 as this is the minimum, you have no established way of choosing "the right component".

File: tests/apps/configurationapp/app.go [hunk 1]
```diff
@@ -36,13 +37,17 @@ import (
...
 const (
 	daprGRPCPort               = 50001
-	configStore                = "configstore"
 	daprConfigurationURLStable = "http://localhost:3500/v1.0/configuration/"
 	daprConfigurationURLAlpha1 = "http://localhost:3500/v1.0-alpha1/configuration/"
 	separator                  = "||"
 	redisHost                  = "dapr-redis-master.dapr-tests.svc.cluster.local:6379"
 	writeTimeout               = 5 * time.Second
 	alpha1Endpoint             = "alpha1"
+	postgresComponent          = "postgres"
```
Comment (artursouza) [inline tests/apps/configurationapp/app.go:46]: Please, keep the test unchanged and switch between stores in the workflow config. See how we do it for Cosmosdb in state store.

**Review by artursouza — CHANGES_REQUESTED** [review]

File: go.mod [hunk 2]
```diff
@@ -245,7 +245,7 @@ require (
 	github.com/influxdata/line-protocol v0.0.0-20210922203350-b1ad95c89adf // indirect
 	github.com/jackc/pgpassfile v1.0.0 // indirect
 	github.com/jackc/pgservicefile v0.0.0-20221227161230-091c0ba34f0a // indirect
-	github.com/jackc/pgx/v5 v5.3.1 // indirect
+	github.com/jackc/pgx/v5 v5.3.1
```
Comment (ItalyPaleAle) [inline go.mod:248]: Make sure that all non-indirect dependencies are in the same block please

File: tests/apps/configurationapp/app.go (hunk 1 above)
Comment (shivamkm07) [reply tests/apps/configurationapp/app.go:46]: Done.

**Comment by shivamkm07** [comment]
@mukundansundar @artursouza ping for reviews.

File: go.mod (hunk 2 above)
Comment (mukundansundar) [reply go.mod:259]: @shivamkm07 small change

File: .github/workflows/kind-e2e.yaml [hunk 3]
```diff
@@ -75,12 +75,15 @@ jobs:
...
           kind-image-sha: sha256:ef453bb7c79f0e3caba88d2067d4196f427794086a7d0df8df4f019d5e336b61
+          DAPR_TEST_CONFIG_STORE: "postgres"
         - k8s-version: v1.24.7
           kind-version: v0.17.0
           kind-image-sha: sha256:577c630ce8e509131eab1aea12c022190978dd2f745aac5eb1fe65c0807eb315
+          DAPR_TEST_CONFIG_STORE: "redis"
         - k8s-version: v1.25.3
           kind-version: v0.17.0
           kind-image-sha: sha256:f52781bc0d7a19fb6c405c2af83abfeb311f130707a0e219175677e366cc45d1
+          DAPR_TEST_CONFIG_STORE: "redis"
```
Comment (artursouza) [inline .github/workflows/kind-e2e.yaml:86]: nit: please, keep the same naming convention. No all caps here.

**Review by artursouza — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (shivamkm07) [reply go.mod]: Done

File: .github/workflows/kind-e2e.yaml (hunk 3 above)
Comment (shivamkm07) [reply .github/workflows/kind-e2e.yaml]: Done

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**